src/python/gudhi/simplex_tree.pyx

```diff
@@ -281,6 +282,8 @@ cdef class SimplexTree:
 
         .. seealso:: :func:`insert_batch`
         """
+        # Without this, it could be slow if we end up inserting vertices in a bad order (flat_map).
+        self.get_ptr().insert_batch_vertices(np.unique(np.stack((edges.row, edges.col))), INFINITY)
```

Comment:
Would it change something to use `np.concatenate` instead of `np.stack`, for example:
```python
row = np.array([2, 5, 3])
col = np.array([1, 4, 6])
dat = np.array([1, 2, 3])
edges = coo_matrix((dat, (row, col)))
edges = coo_matrix((dat, (row, col)))
np.stack((edges.row, edges.col))
# array([[2, 5, 3],
#        [1, 4, 6]], dtype=int32)
np.concatenate((edges.row, edges.col), axis=None)
#array([2, 5, 3, 1, 4, 6], dtype=int32)
```
I know that `np.unique` is doing the job after, but is it faster when it is already a flatten array ?

Reply:
I don't think it makes a difference whether we use stack, concatenate (with or without the axis argument), etc. The only thing that could make a (very small) difference is if one of those functions managed to optimize the case where we start from a 2d array (as in insert_batch) and use arr[0,:] as row and arr[1,:] as col. It would be possible for stack/concatenate to return a reference to arr instead of copying everything to a new array, but I doubt any of those functions tests for this case.

If concatenate looks more natural to you, I can use that, I really picked stack arbitrarily. Maybe it was shorter to type?

Reply:
Ok, let's keep it as is